### PR TITLE
feat(tests): add tests for handling empty array plugin configurations

### PR DESCRIPTION
**Pull request by VarunAthreya**
Add integration tests to verify decK correctly handles plugin configurations with explicitly empty arrays, preventing perpetual diffs that never converge. Tests cover various scenarios top-level and nested empty arrays, enterprise plugins, and string arrays. Also validates that genuine changes from empty to populated arrays are still detected, and that omitted fields match empty arrays.

An unintended side effect was discovered and has been documented here: #2188

### CODE REVIEW

File: tests/integration/diff_test.go [hunk 1]
```diff
@@ -1027,6 +1027,111 @@ func Test_Diff_PluginConfigReorderArraySetValues(t *testing.T) {
...
+	tests := []struct {
+		name      string
+		stateFile string
+		runWhen   func(t *testing.T)
+	}{
+		{
+			// the fixtures set plugin instance_name, which Kong only supports from 3.2.0.
+			name:      "top-level empty array field",
+			stateFile: "testdata/diff/010-empty-array-plugin-config/kong.yaml",
+			runWhen:   func(t *testing.T) { runWhenKongOrKonnect(t, ">=3.2.0") },
```
Comment (Prashansa-K) [inline tests/integration/diff_test.go:1044]: Can you test this in dual mode for Konnect please? 
We can have two test functions if needed - for only Kong and then only Konnect with dual modes. Refer other test files to see how we test against Konnect - with skip defaults and without.

Comment (Prashansa-K) [reply tests/integration/diff_test.go:1044]: Same for other tests

Comment (VarunAthreya) [reply tests/integration/diff_test.go:1044]: done

File: tests/integration/diff_test.go [hunk 2]
```diff
@@ -1119,6 +1148,15 @@ func Test_Diff_EmptyArrayPluginConfig_RealChangeStillDetected(t *testing.T) {
 
 // Test_Diff_EmptyArrayPluginConfig_OmittedFieldMatchesEmptyArray pins down that omitting an array field entirely
 // is indistinguishable from setting it to [], since decK fills both sides with nil before comparing.
+//
+// NOTE: Deliberately NOT wrapped in runDualTestWithSkipDefaults, unlike the two tests above. That helper runs the same
```
Comment (VarunAthreya) [inline tests/integration/diff_test.go:1152]: @Prashansa-K reason for not adding dual test here

should we go ahead with this solution and create a issue to work on this edge case in future or address this now?

**Review by Prashansa-K — APPROVED** [review]

File: tests/integration/diff_test.go (hunk 2 above)
Comment (VarunAthreya) [reply tests/integration/diff_test.go:1152]: created a issue regarding this:
https://github.com/Kong/deck/issues/2188
